### ISSUE TITLE
MultiStepLR Question in DRUNet training

### ISSUE THREAD

**Issue by Akira0703** [cszn/KAIR#101]
In DRUNet training, the scheduler MultiStepLR has milestones of [100000,200000,300000,400000]. My understanding is that the learning rate is halved at each of these "epochs", but your paper says that it is halved at every 100000 ”iterations”. Which is correct?

**Comment by jxtps** [comment]
If you look in the training script they've sort of dispensed with epochs in favor of "steps" or "iters" (= batches processed), presumably in order to be able to stop & then resume the training at any point and just keep going (great for spot instances in the cloud). 

So the MultiStepLR is based on the number of batches you've processed so far, which if you have an infinite dataset is actually a great way to reason about these things. 

(Separately there's some clean up needed in the initialization of the MultiStepLR - they should be feeding in the `current_step` on construction and not into the `scheduler.step(current_step)` call, which should then happen after the optimization, but that's another matter).

**Comment by Akira0703** [comment]
Thank you for your kind explanation.
So you mean that you are optimizing the learning rate which is not related to epochs.
This was helpful as I had a question about setting milestones.
